pkg/logger/otel/provider.go

```diff
@@ -0,0 +1,70 @@
+package otel
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/log"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+
+	"github.com/zondax/golem/pkg/logger"
+)
+
+// createLoggerProvider creates the OpenTelemetry log provider with proper resource configuration
+func (p *Provider) createLoggerProvider(config *logger.OpenTelemetryConfig) (*log.LoggerProvider, error) {
+	exporter, err := p.createExporter(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create OTLP log exporter: %w", err)
+	}
+
+	res := p.createResource(config)
+
+	processor := log.NewBatchProcessor(exporter)
+	provider := log.NewLoggerProvider(
+		log.WithProcessor(processor),
+		log.WithResource(res),
+	)
+
+	return provider, nil
+}
+
+// createResource creates the OpenTelemetry resource with service identification attributes
+// All attributes come from configuration, no environment variables are used
+func (p *Provider) createResource(config *logger.OpenTelemetryConfig) *resource.Resource {
+	attrs := p.buildResourceAttributes(config)
+
+	return resource.NewWithAttributes(
+		semconv.SchemaURL,
+		attrs...,
+	)
+}
+
+// buildResourceAttributes constructs all resource attributes from configuration
+func (p *Provider) buildResourceAttributes(config *logger.OpenTelemetryConfig) []attribute.KeyValue {
+	var attrs []attribute.KeyValue
+
+	// Service name is required
+	if config.ServiceName != "" {
+		attrs = append(attrs, semconv.ServiceName(config.ServiceName))
+	}
```

Comment:
Shouldn't we throw an error here, as the serviceName generates an error if it is empty on the validateConfig function?